source/usr/lib/mcvirt/auth/auth.py

```diff
@@ -0,0 +1,339 @@
+"""Provide auth class for managing permissions."""
+
+# Copyright (c) 2014 - I.T. Dev Ltd
+#
+# This file is part of MCVirt.
+#
+# MCVirt is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MCVirt is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MCVirt.  If not, see <http://www.gnu.org/licenses/>
+
+import os
+import Pyro4
+
+from mcvirt.mcvirt_config import MCVirtConfig
+from mcvirt.exceptions import (UserNotPresentInGroup, InsufficientPermissionsException,
+                               UnprivilegedUserException, InvalidPermissionGroupException,
+                               DuplicatePermissionException)
+from mcvirt.rpc.lock import locking_method
+from mcvirt.rpc.pyro_object import PyroObject
+from mcvirt.auth.permissions import PERMISSIONS, PERMISSION_GROUPS
+
+
+class Auth(PyroObject):
+    """Provides authentication and permissions for performing functions within MCVirt."""
+
+    @staticmethod
+    def check_root_privileges():
+        """Ensure that the user is either running as root
+        or using sudo.
+        """
+        if os.geteuid() == 0:
+            return True
+        else:
+            raise UnprivilegedUserException('MCVirt must be run using sudo')
+
+    def check_user_type(self, *user_type_names):
+        """Check that the currently logged-in user is of a specified type."""
+        if Pyro4.current_context.STARTUP_PERIOD:
+            return True
+
+        user_object = self._get_registered_object('mcvirt_session').get_current_user_object()
+        if user_object.__class__.__name__ in user_type_names:
```

Comment:
Is there a reason for ``user_type_names`` to be a list of class names (as strings) rather than a list of the actual classes? e.g why have `check_user_type('ClusterUser')` instead of importing `ClusterUser` and `check_user_type(ClusterUser)`? It seems safer to me to use the actual classes, because currently you could do something like ``check_user_type('MadeUpUserType')``